### DISCUSSION TITLE
Conda Environment setup problem

### DISCUSSION THREAD

**Discussion by shbkukuk** [libvips/libvips#4217] [General]
I tried install the package with this guide. I tried both way of install Conda and PIP install. But I get this Error 'OSError: cannot load library 'libvips.so.42': libcfitsio.so.5: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libvips.so.42' I checked lib folder in env path. There is not any like file. I tried to install with sudo manager packages but i could not solve the problem. So Do you have any suggestion thansks ?

Ubuntu 20.02
Python : 3.8
Pyvips : 2.1.3
Conda Env

**Comment by vral-parmar** [comment]
Seems you're encountering an issue with missing libraries, specifically `libvips` and `libcfitsio`, which are dependencies of **PyVips**. Here are some steps i resolve this error on **Ubuntu 20.04**:

### 1. **Ensure Required Dependencies Are Installed**
Since you're missing the `libvips` and `libcfitsio` libraries, you can install these dependencies using the package manager.

#### Step 1: Install `libvips`
You can install **libvips** via the package manager:

```bash
sudo apt update
sudo apt install libvips-dev
```

#### Step 2: Install `libcfitsio`
The error also mentions `libcfitsio.so.5`, which is part of the **CFITSIO** library (used for reading/writing astronomical data). Install this dependency with:

```bash
sudo apt install libcfitsio-dev
```

### 2. **Verify the Installation**
After installing both `libvips-dev` and `libcfitsio-dev`, check if the necessary libraries are installed in the correct locations.

```bash
ldconfig -p | grep vips
```

This command should show something like:

```
libvips.so.42 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libvips.so.42
```

### 3. **Check Library Paths**
If you still encounter errors, it could be that the installed libraries are not in your **LD_LIBRARY_PATH**. You can manually add the path where the libraries are located.

```bash
export LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu:$LD_LIBRARY_PATH
```

You can add this to your `~/.bashrc` or `~/.zshrc` for persistence.

### 4. **Reinstall PyVips via PIP**
If the above steps don't resolve the issue, try reinstalling **PyVips**:

```bash
pip uninstall pyvips
pip install pyvips
```

### 5. **Using Conda to Manage Dependencies**
If you're using a **Conda environment**, you can install `libvips` through Conda. This ensures that the dependencies are installed in your environment.

```bash
conda install -c conda-forge pyvips
```

Make sure your **Conda environment** is activated while doing this.

Still im not sure wether its work for you or not, there are so many thing happening behind the scene.
